src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs

```diff
@@ -322,8 +322,28 @@ private async Task MoveSegmentsToProject(string botSegmentsDirectoryPrefix)
                 // So we can safely create this without interfering with the .Move command right below.
                 Directory.CreateDirectory(Path.GetDirectoryName(segmentResourceDirectory));
 
-                // move the directory (this also deletes the source directory)
-                Directory.Move(botSegmentsDirectoryPrefix, segmentResourceDirectory);
+                // If on a Mac, for some weird reason, .Move will fail for project paths that have spaces. We predict
+                // that downstream, something is happening where the space is not properly handled in Unix commands. 
+                // Unfortunately, we could only reproduce this once, but this change should fix it. On Mac, we instead
+                // read and write the files by hand, and avoid any move operations.
+                if (Application.platform == RuntimePlatform.OSXEditor || Application.platform == RuntimePlatform.OSXPlayer)
+                {
+                    Directory.CreateDirectory(segmentResourceDirectory); // Since we aren't moving, we need to create the directory
+                    foreach (var segmentFile in segmentFiles)
+                    {
+                        var sourcePath = botSegmentsDirectoryPrefix + "/" + segmentFile;
+                        var destinationPath = segmentResourceDirectory + "/" + segmentFile;
+                        await File.WriteAllBytesAsync(destinationPath, await File.ReadAllBytesAsync(sourcePath));
```

Comment:
why not use `File.Copy` ?

Reply:
Well that's where I get worried... if Directory.Move isn't working, would File.Copy itself not work? I went with what I thought was the safest option... I am fine with File.Copy but I have no idea if it suffers from the same issue. Technically this too might not be safe.

Reply:
I would definitely use File.Copy as it uses streams and copies small byte buffers at a time instead of reading the full file into memory.  Not that these files are huge, but it is better practice.

It sounds like your real concern is that the recreate is un-reliable.

Reply:
Changed it to File.Copy, but yes my concern is not only that the recreate is unreliable, it's not possible at all 😆 I am making this change but I will not know if this fixes anything until we have the user try it out.

I tried to dig into the Mono source code, but as expected it eventually calls into some native libraries so I can't really inspect **why** Directory.Move wouldn't work, and how different it is from File.Copy. But anyways, this is at least different enough to try out.